src/NextGen/Framework/FormTemplates/Registrars/FormTemplateRegistrar.php

```diff
@@ -0,0 +1,105 @@
+<?php
+
+namespace Give\NextGen\Framework\FormTemplates\Registrars;
+
+use Give\Framework\Exceptions\Primitives\InvalidArgumentException;
+use Give\Framework\PaymentGateways\Exceptions\OverflowException;
+use Give\Log\Log;
+use Give\NextGen\Framework\FormTemplates\FormTemplate;
+
+/**
+ * @unreleased
+ */
+class FormTemplateRegistrar
+{
+    protected $templates = [];
+
+    /**
+     * @unreleased
+     */
+    public function getTemplates(): array
+    {
+        return $this->templates;
+    }
+
+    /**
+     * @unreleased
+     *
+     * @throws InvalidArgumentException
+     */
+    public function getTemplate(string $id): FormTemplate
+    {
+        if (!$this->hasTemplate($id)) {
+            throw new InvalidArgumentException("No template exists with the ID {$id}");
+        }
+
+        /** @var FormTemplate $template */
+        $template = give($this->templates[$id]);
+
+        return $template;
+    }
+
+    /**
+     * @unreleased
+     */
+    public function hasTemplate(string $id): bool
+    {
+        return isset($this->templates[$id]);
+    }
+
+    /**
+     * @unreleased
+     */
+    public function registerTemplate(string $templateClass)
+    {
+        try {
+            $this->register($templateClass);
+        } catch (InvalidArgumentException $invalidArgumentException) {
```

Comment:
What is the reason for catching these two exceptions, specifically, and logging them the same? Should we not just catch a more generic exception?

Reply:
Oh yeah, so I was experimenting with _silently_ failing (not actually throwing an error but logging it) here instead of our usual fatal exceptions.  I've been thinking about this lately for our public api's.  From an application standpoint it makes total sense to throw an exception when warranted. On the other hand, we are a distributed application which makes it potentially fatal to a customer's website (we've seen plenty of exceptions and tiny signature changes break our core functionality).  

For templates specifically, that could mean preventing a donation form from loading - which begs the question.. if there's a template conflict will a log suffice or do we really need to _throw_ an exception?

This is perhaps a longer discussion that id like to open up to the team for thoughts.

Reply:
Let's have that discussion when @JasonTheAdams gets back - I think he'll want input - but go ahead and merge this PR as an incremental step.